node/src/actors/chain_manager/mining.rs

```diff
@@ -229,10 +229,20 @@ impl ChainManager {
 
         for (dr_pointer, data_request_output) in dr_pointers.into_iter().filter_map(|dr_pointer| {
             // Filter data requests that are not in data_request_pool
-            self.chain_state
+            if let Some(dr_state) = self
+                .chain_state
                 .data_request_pool
-                .get_dr_output(&dr_pointer)
-                .map(|data_request_output| (dr_pointer, data_request_output))
+                .data_request_pool
```

Comment:
Why not leave it as before?

```suggestion
                .get_dr_output(&dr_pointer)
```

Reply:
because I need the pkh of the data request. And this field is in DataRequestState but not in DataRequestOutput